fair_mango/dataset/dataset.py

```diff
@@ -58,18 +47,67 @@ def check_real_and_predicted_target_match(
     ValueError
         If the number of real targets and predicted targets does not match.
     """
-    if isinstance(real_target, str) and isinstance(predicted_target, str):
-        return None
-    elif isinstance(real_target, str):
-        if len(predicted_target) != 1:
-            raise ValueError("real_target and predicted_target does not match")
-    elif isinstance(predicted_target, str):
-        if len(real_target) != 1:
-            raise ValueError("real_target and predicted_target does not match")
-    elif len(real_target) != len(predicted_target):
+    if len(real_target) != len(predicted_target):
         raise ValueError("real_target and predicted_target does not match")
 
 
+def validate_columns(
+    sensitive: Sequence[str],
+    real_target: Sequence[str],
+    predicted_target: Sequence[str] | None = None,
+) -> None:
+    """Make sure that the columns provided as parameters are different.
+    A column cannot be a sensitive column and a target at the same time.
+    A column cannot be a real target and a predicted target at the same time.
+
+    Parameters
+    ----------
+    sensitive : Sequence[str]
+        sequence of column names corresponding to sensitive features
+        (Ex: gender, race...).
+    real_target : Sequence[str]
+        sequence of column names corresponding to the real targets
+        (true labels). Every target will be processed independently.
+    predicted_target : Sequence[str], optional
+        sequence of column names corresponding to the predicted targets,
+        by default None
+
+    Raises
+    ------
+    AttributeError
+        if the same column is assigned to different parameters at the same time.
+    """
+    overlap = set(sensitive).intersection(real_target)
+
+    if predicted_target is not None:
+        overlap.update(set(sensitive).intersection(predicted_target))
+        overlap.update(set(real_target).intersection(predicted_target))
+
+    if overlap:
+        raise AttributeError(
+            "Same column name can't be assigned to multiple" f" parameters {overlap}"
+        )
+
+
+def convert_to_list(variable: Sequence[str]) -> Sequence:
+    """Convert a variable of type str to a list
+
+    Parameters
+    ----------
+    variable : Sequence[str]
+        sequence of values.
+
+    Returns
+    -------
+    Sequence
+        sequence of the values (not str).
+    """
+    if isinstance(variable, str):
```

Comment:
impossible case. Fix your typing.

Reply:
I added this function because providing a `str` as a column is possible and not caught although it is typed as `Sequence[str]`.

Reply:
I understand, but I don't think that strings are included in `Sequence[str]`. To fix your type you must have `str | Sequence[str]`, and your return type should be `Sequence[str]`